searchlib/src/vespa/searchlib/engine/trace.h

```diff
@@ -76,17 +82,20 @@ class Trace
     void done();
 
     vespalib::string toString() const;
-    Cursor & getRoot() const { return _root; }
-    vespalib::Slime & getSlime() const { return *_trace; }
+    Cursor * getRoot() const { return _root; }
+    vespalib::Slime * getSlime() const { return _trace.get(); }
     bool shouldTrace(uint32_t level) const { return level <= _level; }
     uint32_t getLevel() const { return _level; }
     Trace & setLevel(uint32_t level) { _level = level; return *this; }
     const RelativeTime & getRelativeTime() const { return _relativeTime; }
 private:
+    void constructObject();
+    void constructTraces();
+    void lazyConstruct(uint32_t level);
```

Comment:
lazy construct should be unconditional and called from functions controlling access to lazy data.

Reply:
Fixed